agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/JvmCompiler.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.microsoft.applicationinsights.agent.bootstrap.diagnostics.PidFinder;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+class JvmCompiler {
+
+  // To disable C2 compiler during Application Insights set-up, for Java >= 9, do the following
+  // things.
+  // Execute with -XX:+UnlockDiagnosticVMOptions -XX:CompilerDirectivesFile=path/jvm-disable-c2.json
+  // Content of jvm-disable-c2.json file:
+  // [
+  //  {
+  //    match: [
+  //      "*::*"
+  //    ],
+  //    c2: {
+  //      Exclude: true
+  //    }
+  //  }
+  // ]
+  private static final String
+      APPLICATIONINSIGHTS_EXPERIMENT_CLEAR_COMPILER_DIRECTIVES_AFTER_INITIALIZATION =
+          "applicationinsights.experiment.clear-compiler-directives-after-initialization";
+
+  private JvmCompiler() {}
+
+  static boolean hasToDisableJvmCompilerDirectives() {
+    return Boolean.getBoolean(
+        APPLICATIONINSIGHTS_EXPERIMENT_CLEAR_COMPILER_DIRECTIVES_AFTER_INITIALIZATION);
```

Comment:
such a long system property name, should we consider a shorter name?

Reply:
this is a hard one to explain succinctly 😅